Data/ByteString/Short/Internal.hs

```diff
@@ -630,3 +1642,57 @@ moduleErrorIO fun msg = throwIO . userError $ moduleErrorMsg fun msg
 
 moduleErrorMsg :: String -> String -> String
 moduleErrorMsg fun msg = "Data.ByteString.Short." ++ fun ++ ':':' ':msg
+
+
+-- Find from the end of the string using predicate.
+--
+-- Return '0' if the predicate returns false for the entire ShortByteString.
+findFromEndUntil :: (Word8 -> Bool) -> ShortByteString -> Int
+findFromEndUntil k sbs = go (length sbs - 1)
+  where
+    ba = asBA sbs
+    w = indexWord8Array ba
+    go !n | n < 0     = 0
+          | k (w n)   = n + 1
+          | otherwise = go (n - 1)
```

Comment:
I think this would be a bit more readable:

 ```suggestion
    go !n | n < 0                    = 0
          | k (indexWord8Array ba n) = n + 1
          | otherwise                = go (n - 1)
```

Reply:
BTW, when `k (indexWord8Array ba n)` is true, why does this function return `n + 1`, and not `n`?!